contracts/ZkBNB.sol

```diff
@@ -261,15 +249,77 @@ contract ZkBNB is UpgradeableMaster, Events, Storage, Config, ReentrancyGuardUpg
         address _nftL1Address,
         uint256 _nftL1TokenId
     ) external {
-        delegateAdditional();
+        requireActive();
+        bytes32 accountNameHash = znsController.getSubnodeNameHash(_accountName);
+        require(znsController.isRegisteredNameHash(accountNameHash), "nr");
+        // Transfer the tokens to this contract
+        bool success;
+        try IERC721(_nftL1Address).safeTransferFrom(
+            msg.sender,
+            address(this),
+            _nftL1TokenId
+        ){
+            success = true;
+        }catch{
+            success = false;
+        }
+        require(success, "ntf");
+        // check owner
+        require(IERC721(_nftL1Address).ownerOf(_nftL1TokenId) == address(this), "i");
+
+        // check if the nft is mint from layer-2
+        bytes32 nftKey = keccak256(abi.encode(_nftL1Address, _nftL1TokenId));
+        uint16 collectionId = 2**16-1;
+        uint40 nftIndex = 2**40-1;
+        uint32 creatorAccountIndex = 2**32-1;
+        uint16 creatorTreasuryRate = 0;
+        bytes32 nftContentHash;
+        uint8 isNewNft = 1;
+        if (l2Nfts[nftKey].nftContentHash == bytes32(0)) {
+            // it means this is a new layer-1 nft
```

Comment:
We have the conclusion not to deposit layer-1 nft?

Reply:
we don't do this in this pr